src/libs/hooks/usePrivateLayooutEffect.js

```diff
@@ -0,0 +1,6 @@
+import { useLayoutEffect, useEffect } from 'react';
+import { isServer } from '../utils';
+
+const usePrivateLayoutEffect = isServer ? useEffect : useLayoutEffect;
```

Comment:
I think it should be called useIsomorphicLayoutEffect

Reply:
@wildergd I agree with @yvmunayev this is a good name